src/main/java/io/github/rabobank/shadow_tool/ShadowFlow.java

```diff
@@ -170,17 +172,29 @@ public Mono<Collection<T>> compareCollections(final Mono<? extends Collection<T>
 
     private void doShadowFlow(final Supplier<Diff> diffSupplier) {
         final var callNewFlow = shouldCallNewFlow();
-        logger.info(MESSAGE_FORMAT, instanceNameLogPrefix, callNewFlow);
+        final var contextMap = MDC.getCopyOfContextMap();
+        logger.info(CALLING_NEW_FLOW, instanceNameLogPrefix, callNewFlow);
 
         if (callNewFlow) {
             try {
-                executorService.submit(() -> logDifferences(diffSupplier.get()));
-            } catch (final Exception e) {
-                logger.error("{} Failed to run the shadow flow", instanceNameLogPrefix, e);
+                executorService.submit(() -> logDifferenceWithMdc(diffSupplier, contextMap));
+            } catch (final Exception t) {
```

Comment:
Exception t?

Reply:
Interesting, I honestly don't know where that came from. I'll change it to a more descriptive `e`.

Reply:
Much better!